### PR TITLE
Feat/popups

### PR DESCRIPTION
**Pull request by denisehung**
Aloha! I created the following popups, all based on PopupWithForm:

- SignIn
- SignUp
- SuccessPopup (on successful registration)

I already linked the SignIn popup to the Header button. And when you click on the 'sign in' or 'sign up' link at the bottom of the form, the corresponding popup will open.

### CODE REVIEW

File: src/components/signIn/SignIn.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import PopupWithForm from '../popupWithForm/PopupWithForm';
+
+function SignIn(props) {
+  return(
+    <PopupWithForm name="sign-in" title="Sign in" isOpen={props.isOpen} onClose={props.onClose} isLoading={props.isLoading}>
+
+      <div className="popup__input-wrapper">
```
Comment (cjmaret) [inline src/components/signIn/SignIn.js:8]: could you have put the email and password sections in PopupWithForm, that way you'd only have to write them once?

File: src/components/signIn/SignIn.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <label className="popup__input-label" for="email-input">Email</label>
```
Comment (cjmaret) [inline src/components/signIn/SignIn.js:9]: the console said these ought to be `htmlFor`

**Review by cjmaret — CHANGES_REQUESTED** [review]
like it! you even got the ESC function and 'click outside'.

I think the for attributes on the labels should be changed to 'htmlFor'. and as for the other comment about the email and password, we don't have to change it if you dont want to, it works fine as is.

File: src/components/signIn/SignIn.js (hunk 1 above)
Comment (denisehung) [reply src/components/signIn/SignIn.js]: Good point! I've changed it

File: src/components/signIn/SignIn.js (hunk 2 above)
Comment (denisehung) [reply src/components/signIn/SignIn.js]: Changed these too

**Comment by denisehung** [comment]
Aloha! I made the requested changes and improved the css a bit. I'm only not sure about the naming structure of the folders yet, so I kept that as it is (for now).

**Comment by denisehung** [comment]
I added the submit functionalities to the popups, so when you sign in now, the header will change accordingly.
Unfortunately, I had to place the input fields back to the corresponding components (SignUp and SignIn) because otherwise I didn't know how to reset them all after submitting. I think my code can be more efficient too, but for now I just wanted to make it work ;)

**Comment by cjmaret** [comment]
looks good!

for the footer, i notice clicking 'home' adds '#home' to the URL... just a reminder that we have to fix that in the future, i dont remember how

**Pull request merged**